### PR TITLE
[improve](CI)Core modules require maintainer review

### PR DESCRIPTION
**Pull request by caoliang-web**
## Proposed changes

Configuration File Documentation
This configuration file is used to set maintainers for each path in a code repository. Only maintainers of a path can approve code changes under that path before they can be merged.

Configuration File Format

The configuration file is a JSON file that contains a paths array. Each element in the array represents a path and its list of maintainers.

Example

```
{
  "paths": [
    {
      "path": "docs/*",
      "maintainers": [
        "luzhijing",
        "morningman"
      ]
    }
  ]
}
```

Configurationの

paths: An array of objects that represent paths and their maintainers.
path: A string that represents the path that maintainers need to be set for. You can use the wildcard * to match multiple paths.
maintainers: An array of strings that represents the Github usernames  of the maintainers for the path.
Instructions

Add the configuration file to the root directory of your code repository.
In your code repository's settings, enable "Require pull request reviews before merging".
In your code repository's settings, set "Required approving reviewers" to the maintainers specified in the configuration file.

If a path does not have any maintainers set, then anyone can approve code changes under that path.

Suppose your code repository has the following file structure:

docs/
├── [index.md](http://index.md/)
└── [README.md](http://readme.md/)
And your configuration file is as follows:

```
{
  "paths": [
    {
      "path": "docs/*",
      "maintainers": [
        "luzhijing",
        "morningman"
      ]
    }
  ]
}
```

In this case, only Lu Zhijing and Morning Man can approve code changes under the docs/ directory.
Further Information

You can use multiple configuration files to set maintainers for different paths.
You can use GitHub Actions to automate the code review and merging process.

### CODE REVIEW

**Review by smallhibiscus — APPROVED** [review]

**Review by smallhibiscus — APPROVED** [review]

**Review by smallhibiscus — APPROVED** [review]

**Review by gnehil — APPROVED** [review]

File: .github/workflows/pr-approve-status.yml [hunk 1]
```diff
@@ -19,42 +19,25 @@ name: Need_2_Approval
 
 on:
   pull_request_review:
-    types: [submitted]
+    types: [ submitted ]
```
Comment (gnehil) [inline .github/workflows/pr-approve-status.yml:22]: test

**Review by gnehil — CHANGES_REQUESTED** [review]

File: .github/workflows/pr-approve-status.yml [hunk 2]
```diff
@@ -19,42 +19,25 @@ name: Need_2_Approval
...
-              echo "PR ${pr_num} has not been approved by at least 2 reviewers"
-              # shellcheck disable=SC2242
-              exit 1
-          fi
-          echo "Thanks for your contribution to Doris."
-          exit 0
+      - name: Install Python dependencies
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.10'  # Adjust if needed
```
Comment (smallhibiscus) [inline .github/workflows/pr-approve-status.yml:33]: test

**Review by smallhibiscus — CHANGES_REQUESTED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Comment by dataroaring** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Comment by dataroaring** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**